### ISSUE TITLE
[Feature]: Include logs from multiple processes in one run

### ISSUE THREAD

**Issue by zplizzi** [wandb/wandb#3766]
### Description

I have a training setup that involves multiple processes running simultaneously (i'm doing RL, so I have processes playing a game, a process training the model, processes loading data, etc). I would like to collect metrics from all of them, and be able to see them all in the same wandb run. I don't think this is possible currently, afaik? I've seen y'all suggest calling `wandb.init()` in each process and grouping them all together, but that seems awfully messy.

### Suggested Solution

^

**Comment by MBakirWB** [comment]
@zplizzi ,

That is correct,  parallel processes will be logged into separate different "runs" that can be grouped into a single experiment. They cannot be merged together into a single run . There are three methods to set grouping, see [here](https://docs.wandb.ai/guides/track/advanced/grouping#use-cases). Setting the group parameter in `wandb.init() `in your script, setting a group environment variable using `WANDB_RUN_GROUP` , or by toggling grouping in the `WebUI`. Can you please provide me with your specific use case requirements and how this feature could help your work. I will add your notes to the ticket. 

Regards,

Mohammad

**Comment by zplizzi** [comment]
Let me know if the use case I described above isn't clear.

Using groups to solve this problem works ok, but has some limitations. Most obviously, I can now no longer use the group feature for what it was actually intended for (grouping multiple logically independent runs together for analysis), if I'm already using it to combine processes within one run. But also if I'm using groups in this way, then I basically have to use the project page to view the run data (since any individual run in the group will only have part of the data). And the project page doesn't display all the data in the same way - for example, histograms don't show up on the project page (not by default, at least?), and images+videos show up small. Other things are less convenient - for example, I commonly rename runs during or after a run for organization purposes (eg I'll name a run by how it performed, which is useful making graph legends that make sense). There's no way to change the name of a group after it's been created, so I can't do that anymore. I'm sure I'll run into more similar limitations as I work with this more.
